### ISSUE TITLE
Add a setting to add "NOLOCK" to all "SELECT"

### ISSUE THREAD

**Issue by AndreiMaz** [nopSolutions/nopCommerce#6899]
Let's add a setting to appsettings.json file indicating whether linq2db should add "NOLOCK" to all "SELECT". Let's name this setting "UseNolock" (disabled by default)

For example, src/Libraries/Nop.Data/DataProviders/MsSqlDataProvider.cs:

```
public override IQueryable<TEntity> GetTable<TEntity>()
        {
            return new DataContext(LinqToDbDataProvider, GetCurrentConnectionString())
            {
                MappingSchema = GetMappingSchema(),
                CommandTimeout = DataSettingsManager.GetSqlCommandTimeout()
            }
                .GetTable<TEntity>().AsSqlServer().WithNoLock();
        }
```

**Comment by caglarsarikaya** [comment]
Doesn't it reduce the accuracy of submitted queries?

**Comment by glhays** [comment]
With a setting the choice be yours, so depending on what side of the nolock
opinion you fall. Turn it on or off. I see no harm in having choices.

**Comment by ahmadkq** [comment]
@exileDev is there a way to add this to Nop4.40? any idea how can i do it?

**Comment by exileDev** [comment]
Closed #6899

**Comment by exileDev** [comment]
Closed https://github.com/nopSolutions/nopCommerce/issues/6899
